fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/BaseCompositeAvailability.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+import com.yahoo.bard.webservice.util.StreamUtils;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A class implementing common capabilities for availabilities backed by a collection of other availabilities.
+ */
+public abstract class BaseCompositeAvailability implements Availability {
+
+    private final Set<Availability> sourceAvailabilities;
+    private final Set<TableName> dataSourcesNames;
+
+    /**
+     * Constructor.
+     *
+     * @param availabilityStream  A potentially ordered stream of availabilities which supply this composite view
+     */
+    protected BaseCompositeAvailability(Stream<Availability> availabilityStream) {
```

Comment:
I don't especially like Streams for public APIs, but since this is a protected constructor on an abstract class I thought we could be more tightly tuned.